tests/numerics/fv/test_tpsa.py

```diff
@@ -21,1418 +36,1426 @@
 
-class TestTpsaTailoredGrid:
-    """Define a grid of two cells, one internal face. Verify that the discretization
-    matrices are constructed correctly for the internal face, as well as two boundary
-    faces for Dirichlet, Neumann and Robin conditions.
+    if tensile:
+        # Positive x-direction, negative y- (and z-) direction.
+        assert np.all(x[: g.dim * g.num_cells : g.dim] > 0)
+        assert np.all(x[1 : g.dim * g.num_cells : g.dim] < 0)
+        if g.dim == 3:
+            assert np.all(x[2 : g.dim * g.num_cells : g.dim] < 0)
 
-    The test strategy is to compare the discretization with known values that are
-    computed by hand following the description of the Tpsa discretization provided in
-    the paper. This should mainly be straightforward, though there is room for confusion
-    e.g. in sign conventions for normal vectors. Experience shows, however, that such
-    errors are picked up by convergence tests and similar.
+        # Total pressure should be positive (expansion).
+        assert np.all(x[(g.dim + rot_dim) * g.num_cells :] > 0)
+    else:
+        assert np.all(x[: g.dim * g.num_cells : g.dim] < 0)
+        assert np.all(x[1 : g.dim * g.num_cells : g.dim] > 0)
+        if g.dim == 3:
+            assert np.all(x[2 : g.dim * g.num_cells : g.dim] > 0)
+
+        assert np.all(x[(g.dim + rot_dim) * g.num_cells :] < 0)
 
-    Scope of difficulty for the tests (described here for future reference in case we
-    feel like reassessing the test coverage): The target faces have indices 1 (the
-    internal one), 0 (boundary face, normal vector along x-axis, normal vector points
-    into the cell, unit area of face, unit distance between cell and face center), and 6
-    (boundary face, not aligned with the axis, non-unitary area and cell-face distance,
-    outward-pointing normal vector). The Lame and Cosserat parameters are heterogeneous.
 
+@pytest.mark.parametrize("g", [pp.CartGrid([2, 2]), pp.CartGrid([2, 2, 2])])
+def test_translation(g: pp.Grid):
+    """Set boundary conditions that correspond to a translation of the grid, check that
+    the interior cells follow the translation, and that the resulting system is
+    stress-free.
     """
+    g.compute_geometry()
 
-    @pytest.fixture(autouse=True)
-    def setup(self):
-        # Define grid.
-        g = pp.CartGrid([2, 1])
-        g.nodes = np.array(
-            [[0, 0, 0], [2, 0, 0], [3, 0, 0], [0, 1, 0], [1, 2, 0], [3, 1, 0]]
-        ).T
-        g.compute_geometry()
-        g.face_centers[0, 3] = 1.5
-        g.cell_centers = np.array([[1, 0.5, 0], [2.5, 0.5, 0]]).T
-        self.g = g
+    d = _set_uniform_parameters(g)
 
-        # Define shear modulus and Cosserat parameter.
-        self.mu_0 = 1
-        self.mu_1 = 2
-        self.cos_0 = 1
-        self.cos_1 = 3
+    # Set type and values of boundary conditions. No rotation.
+    _set_uniform_bc(g, d, "dir")
+    if g.dim == 2:
```

Comment:
You could just use the 3d in both cases.

Reply:
Thanks.